core/templates/dev/head/components/share/SharingLinksDirective.js

```diff
@@ -22,39 +22,51 @@ oppia.directive('sharingLinks', [function() {
     scope: {
       layoutType: '@',
       layoutAlignType: '@',
+      shareType: '@',
       getTwitterText: '&twitterText',
-      getExplorationId: '&explorationId'
+      getExplorationId: '&explorationId',
+      getCollectionId: '&collectionId'
     },
     templateUrl: 'components/sharingLinks',
     controller: [
-      '$scope', '$window', 'oppiaHtmlEscaper', 'ExplorationEmbedButtonService',
+      '$scope', '$window', '$injector', 'oppiaHtmlEscaper',
```

Comment:
'ExplorationEmbedButtonService', is needed only when directive is in exploration and not required when directive is in collection.
So, $injector is needed to get ExplorationEmbedButtonService when directive used in exploration.
It is used in line 39